### PR TITLE
Added encoding for paths

### PR DESCRIPTION
**Pull request by grbIzl**
- Encoding added. For Windows path local codepage will be used. For posix it will be usual utf-8 string
- Test with non unicode path added

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,3 +26,4 @@ path = "test/test.rs"
 [dependencies]
 libc = "0.2"
 rocksdb-sys = { path = "rocksdb-sys", version = "0.3.0" }
+local-encoding = "*"
```
Comment (rphmeier) [inline Cargo.toml]: should pin to specific version

File: src/lib.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 //
 extern crate rocksdb_sys;
+extern crate local_encoding;
```
Comment (rphmeier) [inline src/lib.rs]: style: newline between links and pub exports

File: src/rocksdb.rs [hunk 3]
```diff
@@ -276,7 +278,8 @@ impl DB {
     	if cfs.len() != cf_opts.len() {
 			return Err(format!("Mismatching number of CF options"));
 		}
-        let cpath = match CString::new(path.as_bytes()) {
+        let encoded_path = Encoding::ANSI.to_bytes(path);
+        let cpath = match CString::new(encoded_path.unwrap()) {
```
Comment (rphmeier) [inline src/rocksdb.rs]: `unwrap()` is an assertion. Favor `expect` which lets you provide a proof for why the assertion will never fail

Comment (rphmeier) [inline src/rocksdb.rs]: (our goal is zero panics in Parity code, so all panickers should be backed by a "proof" of why they won't fail)

File: src/rocksdb.rs [hunk 4]
```diff
@@ -378,7 +381,9 @@ impl DB {
     }
 
     pub fn destroy(opts: &Options, path: &str) -> Result<(), String> {
-        let cpath = CString::new(path.as_bytes()).unwrap();
+        let encoded_path = Encoding::ANSI.to_bytes(path);
+        let cpath = CString::new(encoded_path.unwrap()).unwrap();
```
Comment (rphmeier) [inline src/rocksdb.rs]: fix unwrap

File: src/rocksdb.rs [hunk 5]
```diff
@@ -395,7 +400,9 @@ impl DB {
     }
 
     pub fn repair(opts: &Options, path: &str) -> Result<(), String> {
-        let cpath = CString::new(path.as_bytes()).unwrap();
+        let encoded_path = Encoding::ANSI.to_bytes(path);
+        let cpath = CString::new(encoded_path.unwrap()).unwrap();
```
Comment (rphmeier) [inline src/rocksdb.rs]: fix unwrap

File: src/rocksdb.rs [hunk 6]
```diff
@@ -518,7 +525,8 @@ impl DB {
                      name: &str,
                      opts: &Options)
                      -> Result<Column, String> {
-        let cname = match CString::new(name.as_bytes()) {
+        let encoded_name = Encoding::ANSI.to_bytes(name);
+        let cname = match CString::new(encoded_name.unwrap()) {
```
Comment (rphmeier) [inline src/rocksdb.rs]: fix unwrap

**Comment by grbIzl** [comment]
Review comments fixed

File: src/rocksdb.rs [hunk 7]
```diff
@@ -276,7 +278,16 @@ impl DB {
     	if cfs.len() != cf_opts.len() {
 			return Err(format!("Mismatching number of CF options"));
 		}
-        let cpath = match CString::new(path.as_bytes()) {
+        let encoded_path = match Encoding::ANSI.to_bytes(path) {
```
Comment (rphmeier) [inline src/rocksdb.rs]: idiomatic rust tip: you can use `map_err` and the `?` operator to condense this to
`let encoded_path = Encoding::ANSI.to_bytes(path).map_err(|_| "error msg".to_string())?`

**Comment by rphmeier** [comment]
LGTM, thanks

**Pull request merged**